api/v1beta2/common.go

```diff
@@ -19,38 +19,29 @@ package v1beta2
 import (
 	"strconv"
 
+	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 )
 
 func defaultIBMPowerVSMachineSpec(spec *IBMPowerVSMachineSpec) {
-	if spec.Memory == "" {
-		spec.Memory = "4"
+	if spec.MemoryGiB == 0 {
+		spec.MemoryGiB = 4
 	}
-	if spec.Processors == "" {
-		spec.Processors = "0.25"
+	if spec.Processors.StrVal == "" && spec.Processors.IntVal == 0 {
+		spec.Processors = intstr.FromString("0.5")
 	}
-	if spec.SysType == "" {
-		spec.SysType = "s922"
+	if spec.SystemType == "" {
+		spec.SystemType = "s922"
 	}
-	if spec.ProcType == "" {
-		spec.ProcType = "shared"
+	if spec.ProcessorType == "" {
+		spec.ProcessorType = PowerVSProcessorTypeShared
 	}
 }
 
-func validateIBMPowerVSSysType(spec IBMPowerVSMachineSpec) (bool, IBMPowerVSMachineSpec) {
-	sysTypes := [...]string{"s922", "e980"}
-	for _, st := range sysTypes {
-		if spec.SysType == st {
-			return true, IBMPowerVSMachineSpec{}
-		}
-	}
-	return false, spec
-}
-
-func validateIBMPowerVSProcType(spec IBMPowerVSMachineSpec) (bool, IBMPowerVSMachineSpec) {
-	procTypes := [...]string{"shared", "dedicated", "capped"}
-	for _, pt := range procTypes {
-		if spec.ProcType == pt {
+func validateIBMPowerVSSystemType(spec IBMPowerVSMachineSpec) (bool, IBMPowerVSMachineSpec) {
```

Comment:
do we want to have this check? because there are some datacenters got some other system type as well and also this list is different for the staging environment, IMO: it is good to avoid the validation for the system type.

Reply:
Removed this check, added enum validation for now (to avoid unexpected values).